query-service-impl/src/main/java/org/hypertrace/core/query/service/prometheus/PrometheusMetricQueryResponseParser.java

```diff
@@ -0,0 +1,162 @@
+package org.hypertrace.core.query.service.prometheus;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.TypeAdapter;
+import com.google.gson.stream.JsonReader;
+import com.google.gson.stream.JsonWriter;
+import java.io.IOException;
+
+public class PrometheusMetricQueryResponseParser {
+
+  public static PrometheusMetricQueryResponse parse(String jsonString) {
```

Comment:
I did this way as I had to parse value array which is timestamp and string in response (https://prometheus.io/docs/prometheus/latest/querying/api/#instant-queries) So, thought of organizing data a bit and so parsed fully. 

Pl. let me know if you guys are thinking differently.

Reply:
Moved to Jackson-based parser using object mapper for simplicity (mapped to the same set of classes required from API).